web/src/narrow_state.ts

```diff
@@ -349,7 +349,7 @@ export function narrowed_by_pm_reply(current_filter: Filter | undefined = filter
     if (current_filter === undefined) {
         return false;
     }
-    const terms = current_filter.terms();
+    const terms = current_filter.terms().filter((term) => term.operator !== "with");
     return terms.length === 1 && current_filter.has_operator("dm");
 }
```

Comment:
Should the `near` operator be handled here too? Overall I'm worrying that we're not processing `near/with` operators in the right places; maybe those should not be stored as `terms` on the filter object.